brave/src/main/java/brave/internal/handler/NoopAwareFinishedSpanHandler.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright 2013-2019 The OpenZipkin Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package brave.internal.handler;
+
+import brave.handler.FinishedSpanHandler;
+import brave.handler.MutableSpan;
+import brave.internal.Platform;
+import brave.propagation.TraceContext;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicBoolean;
+import zipkin2.Call;
+
+/**
+ * When {@code noop}, this drops input spans by returning false. Otherwise, it logs exceptions
+ * instead of raising an error, as the supplied handler could have bugs.
+ */
+public abstract class NoopAwareFinishedSpanHandler extends FinishedSpanHandler {
+  public static FinishedSpanHandler create(List<FinishedSpanHandler> handlers, AtomicBoolean noop) {
+    if (handlers.isEmpty()) return FinishedSpanHandler.NOOP;
+
+    if (handlers.size() == 1) {
+      FinishedSpanHandler onlyHandler = handlers.get(0);
+      return onlyHandler == FinishedSpanHandler.NOOP ? onlyHandler : new Single(onlyHandler, noop);
+    }
+
+    int i = 0;
+    boolean alwaysSampleLocal = false, supportsOrphans = false;
+    FinishedSpanHandler[] copy = new FinishedSpanHandler[handlers.size()];
```

Comment:
May as well use toArray to make the copy and avoid the mysterious `i`

Reply:
yeah good idea.. this was more drift from previous versions than anything else.